build/teamcity/internal/release/process/build-cockroach-release-per-platform.sh

```diff
@@ -54,7 +54,11 @@ export google_credentials="$gcs_credentials"
 source "build/teamcity-support.sh"  # For log_into_gcloud
 log_into_gcloud
 export GOOGLE_APPLICATION_CREDENTIALS="$PWD/.google-credentials.json"
-$BAZEL_BIN/pkg/cmd/publish-provisional-artifacts/publish-provisional-artifacts_/publish-provisional-artifacts -provisional -release --gcs-bucket="$gcs_bucket" --output-directory=artifacts --platform=$platform
+cat licenses/THIRD-PARTY-NOTICES.txt > /tmp/THIRD-PARTY-NOTICES.txt.tmp
+echo "Additional licenses" >> /tmp/THIRD-PARTY-NOTICES.txt.tmp 
+echo >> /tmp/THIRD-PARTY-NOTICES.txt.tmp
+cat $(ls -1 licenses/* | grep -v THIRD-PARTY-NOTICES.txt | sort ) >> /tmp/THIRD-PARTY-NOTICES.txt.tmp
```

Comment:
We need to do something more here. Just concatenating the licenses together makes it impossible to distinguish what a given license is related to. It also makes it difficult to determine where one license ends and the next begins. The something more could be one or multiple of:

- Add header information to each of the files in the `licenses/` directory with information about what the license is. This header information would be similar to the header information included in the base THIRD-PARTY-NOTICES.txt file for each dependency.
    - This is probably the best path to take as it gives us the most control about what meta information to include before the file's notice contents. If we follow this path, then we'd also want to add a few blank lines between files because most files do not include blank lines at the end of them.
- Change this to append a separator between files.
    - Good to at least see where one file begins and ends.
- Append the source file name before appending the contents of the file.
    - This would work to give some context, but the filename can't include much information and it often needs to be adjusted (ex: to remove `/`s from paths).